### DISCUSSION TITLE
reverse proxy behind path

### DISCUSSION THREAD

**Discussion by hmoffatt** [semaphoreui/semaphore#1510] [Q&A] [answered]
Is it supported to deploy semaphore behind a reverse proxy to a path other than /, ie https://example.com/semaphore?

It's not mentioned in the reverse proxy documentation https://docs.ansible-semaphore.com/administration-guide/security

**Comment by hmoffatt** [comment]
There is such a setting: https://docs.ansible-semaphore.com/administration-guide/configuration#public-url

However, it doesn't seem to work. Attempts to fetch the js/css assets get redirected to an invalid path.

**Reply by beKGB** [reply]
it is possible, on my machine the following address works: https://myHost/semaphore

**Reply by hmoffatt** [reply]
Yes I figured out that it changes the base URL of the internal HTTP server as well, so you have to map outside /semaphore to eg http://localhost:3000/semaphore not just http://localhost:3000/

However it breaks OpenID Connect so I am not using the path for now.

**Comment by beKGB** [comment]
post nginx and docker compose config

**Comment by hmoffatt** [comment, marked as answer]
The web host parameter changes the base URL of the internal HTTP server as well, so you have to map outside /semaphore to eg http://localhost:3000/semaphore not just http://localhost:3000/

Here's what I got working with Apache httpd:

```
	ProxyPass /semaphore/api/ws ws://localhost:5010/semaphore/api/ws retry=0
	ProxyPass /semaphore http://localhost:5010/semaphore retry=0
```

However, OpenID Connect is not working with the web path configured.

**Reply by beKGB** [reply]
i have https (ngins/proxy), harbor and semaphore and it works for me, 

you need to properly configure redirects on apache :
(port 5010 if semaphore on 5010)
ProxyPass /semaphore/api/ws ws://localhost:5010/api/ws 
ProxyPass /semaphore http://localhost:5010/

**Comment by hmoffatt** [comment]
> ProxyPass /semaphore/api/ws ws://localhost:5010/api/ws
> ProxyPass /semaphore http://localhost:5010/

Are you sure about this? It does not work for me, all of the endpoints return 404. It should be

ProxyPass /semaphore/api/ws ws://localhost:5010/semaphore/api/ws
ProxyPass /semaphore http://localhost:5010/semaphore

However, the OpenID Connect endpoints do not work. The redirects do not take into account the base URL.

**Reply by beKGB** [reply]
i have:
  semaphore:
    container_name: semaphore
    restart: always
    ports:
      - 3000:3000


if you have port: 5010, ok 

but you must set this port in the semaphore config.json (3000 is default)
config.json:
....
"port": "5010",
"web_host": "https://yourdomain.com/semaphore",
.....


next , i have nginx-proxy, and i set:

  upstream semaphore {
    server semaphore:3000;
  }

and


    location /semaphore/api/ws {
      proxy_pass http://semaphore/api/ws;
      proxy_http_version 1.1;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection "upgrade";
      proxy_set_header Origin "";
    }

in the above:

http://semaphore => points to :   upstream semaphore {
    server semaphore:3000;
  }

**Reply by hmoffatt** [reply]
What did you set the web root to in the Semaphore config? Which version do you use?

**Reply by beKGB** [reply]
v2.9.35 (latest)
![image](https://github.com/ansible-semaphore/semaphore/assets/31951846/b262aa01-0582-454b-a558-c8c2ea92ff1b)
